### PR TITLE
v2 sdk

### PR DESCRIPTION
**Pull request by marktoda**
- v2 order builder
- enable pre-sign flow
- use original value if overrides == 0
- cosignerData or default
- typo
- fix v2order test

### CODE REVIEW

File: src/order/V2DutchOrder.ts [hunk 1]
```diff
@@ -40,9 +41,17 @@ export type CosignerDataJSON = {
...
 export type V2DutchOrderInfo = OrderInfo & {
   cosigner: string;
-  cosignerData: CosignerData;
   input: DutchInput;
   outputs: DutchOutput[];
+  cosignerData?: CosignerData;
+  cosignature?: string;
+};
+
+export type CosignedV2DutchOrderInfo = OrderInfo & {
```
Comment (zhongeric) [inline src/order/V2DutchOrder.ts:50]: nit: `Omit<V2DutchOrderInfo, 'cosignerData', 'cosignature' & { cosignerData: CosignerData; cosignature: string; }` ?

File: src/order/V2DutchOrder.ts [hunk 2]
```diff
@@ -327,11 +347,14 @@ export class V2DutchOrder extends V2Order {
...
-   * @inheritdoc Order
+   *  recovers co-signer address from cosignature and full order hash
+   *  @param fullOrderHash The full order hash over (orderHash || cosignerData)
+   *  @param cosignature The cosignature to recover
+   *  @returns The address which co-signed the order
    */
   recoverCosigner(
     fullOrderHash: string,
-    cosignature: string = this.info.cosignature
+    cosignature: string = this.info.cosignature ?? "0x"
```
Comment (zhongeric) [inline src/order/V2DutchOrder.ts:357]: any reason to add a default value here?

**Review by zhongeric — APPROVED** [review]

**Comment by marktoda** [comment]
TY @zhongeric will handle in a followup

File: src/order/V2DutchOrder.ts (hunk 2 above)
Comment (marktoda) [reply src/order/V2DutchOrder.ts:357]: I think because the cosignature is not there after the first stage of the auction

**Comment by marktoda** [comment]
https://github.com/Uniswap/uniswapx-sdk/pull/115

**Pull request merged**